docs/adr/adr-007-data-retention.md

```diff
@@ -0,0 +1,196 @@
+
+## Decision
+
+We decided that historic, unused data should be removed periodically from the
+system entirely. It was concluded that the data the system no longer uses
+should only exist for a maximum of 1 year before removal. We sometimes
+analyse historic data from up to a year ago but we have no evidence of needing
+this data for analytics beyond a year.
+
+Emails (and their associated SubscriptionContents) are the one place where we
+already manage retention - these are held locally for 7 days, and also
+archived to Athena indefinitely. We still need this low retention for Emails in
+order to manage the size of the table. However, the archiving to Athena is
+inconsistent with the above decision to delete unused data entirely. While we
+originally thought Athena might serve as a data source for analytics, this has
+not happened in practice, so the archive data stored there is entirely unused.
+We therefore decided to remove the archiving mechanism for Emails, since it has
+no value.
+
+The following list documents our approach to the decision for different data
+in the system.
+
+* **ContentChange, Message & DigestRun**. Remove once their `created_at` value is
+  over a year old.
+
+* **MatchedContentChange, MatchedMessage & DigestRunSubscriber**. Cascaded when
+  their parent content change / message / digest run is deleted.
+
+* **Subscription**. Remove subscriptions when they've been ended for over a
+  year, based on their `ended_at` value. Once a subscription is ended it [can't
+  currently be made active again] and so can be deleted.
+
+* **Previously subscribed to SubscriberList**. Remove lists when they have no
+  active subscriptions and their latest, ended subscription is over a year old.
+  It is possible for a list we deem as historic to be subscribed to again,
+  however this seems unlikely due to the list not having any active
+  subscriptions for over a year.  If this was to happen, after the list has
+  been deleted then a new `SubscriberList` record will be generated.
+
+* **Unused SubscriberList**. Remove lists which have never had any
+  subscriptions and are over 7 days old. A scenario of when this can happen is
+  when a user quits their signup journey midway through. 7 days was the chosen
+  retention period to allow users to complete their signup journey as they have
+  a [7 day limit] to confirm their subscription. As mentioned in the context
+  section, unused subscriber lists pollute our analytics and so we should
+  remove them regularly.
+
+* **Subscriber**. Remove subscribers once their `created_at` is over a year old
+  and they don't have any subscriptions (active or ended). Subscribers should
+  only be without subscriptions when their historic subscriptions have been
+  deleted. This means we should be able to delete all subscribers (under a year
+  old too) that don't have any subscriptions. However, we should be cautious
```

Comment:
Minor - this area of caution seems somewhat superfluous to the decision. I'd suggest stopping the paragraph at "subscriptions."

Reply:
I've reworded the sentence so it relates more to the decision (as it was part of the decision). Let me know if thats any better?